### ISSUE TITLE
`firebase serve` not working properly on macOS

### ISSUE THREAD

**Issue by paulkre** [firebase/firebase-tools#1675]
### [REQUIRED] Environment info

**firebase-tools:** 7.3.2

**Platform:** macOS

### [REQUIRED] Test case

When serving any firebase project on macOS, [http://localhost:5000](http://localhost:5000) will respond with an unidentified binary file. Visiting [http://localhost:5000/index.html](http://localhost:5000/index.html) works as expected.

### [REQUIRED] Steps to reproduce

Initialise a new project by running `firebase init`, select "Hosting" and choose all default options.  
After the initialisation, first run `firebase serve` and in a new window run `curl -i http://localhost:5000`.  

### [REQUIRED] Expected behavior

```bash
$ curl -i http://localhost:5000
HTTP/1.1 200 OK
Content-Type: text/html; charset=utf-8
Content-Length: 3505
ETag: 600c578796efad9d567f407af9846c06
Last-Modified: Wed, 25 Sep 2019 12:52:11 GMT
Date: Wed, 25 Sep 2019 16:28:01 GMT
Connection: keep-alive

<!DOCTYPE html>
<html>
  <head>
...
```

### [REQUIRED] Actual behavior

```bash
$ curl -i http://localhost:5000
HTTP/1.1 200 OK
Content-Type: false
Content-Length: 131072
ETag: d9f3ac97476078d9d84ea9a43d38c28d
Last-Modified: Wed, 25 Sep 2019 12:52:22 GMT
Date: Wed, 25 Sep 2019 16:21:16 GMT
Connection: keep-alive

�, �V9O�V9O�n9O�������[W�404.html�
m �V9O�V9OBp9O������
�index.html%
```

**Comment by samtstern** [comment]
@paulkre that is very strange.  Can you show me the contents of your `firebase.json` file and also your `index.html` file?

**Comment by bkendall** [comment]
I'm unable to replicate this on my machine using the given steps.

Since it's returning something _entirely_ different (it looks like it's listing out the directory, which makes little sense), there might be something in the environment that's doing something odd.

1. What's the version of NodeJS you are using?
1. Is there something else running on `localhost:5000`?
1. How did you get the expected behavior output? 🙂

**Comment by paulkre** [comment]
Thanks for the replies. I just realised, the problem only occurs when I serve a project located on an external drive (`/Volumes/...`).  If the project is located on my internal drive, everything works as expected.

@samtstern  I just used the most basic project you get when you run `firebase init`.

@bkendall My NodeJS version is v10.15.1, I have nothing else running on port 5000 and I got the correct output response by running `http://localhost:5000/index.html` which works on all drives.
